File: problems/Valid Anagram/isAnagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) {
            return false;
        }

        var store = new int[26];

        for (var i = 0; s.Length > i; ++i) {
            ++store[s[i] - 'a'];
            --store[t[i] - 'a'];
        }

        foreach (var item in store) {
            if (0 != item) {
                return false;
            }
        }

        return true;
    }
}

File: problems/Valid Anagram/isAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        int[] letterFrequencies = new int[26];

        for (int idx = 0; s.length() > idx; ++idx) {
            ++letterFrequencies[s.charAt(idx) - 'a'];
            --letterFrequencies[t.charAt(idx) - 'a'];
        }

        for (int item : letterFrequencies) {
            if (0 != item) {
                return false;
            }
        }

        return true;
    }
}
